iroh-net/src/relay/codec.rs

```diff
@@ -264,6 +264,57 @@ impl Frame {
         }
     }
 
+    fn from_ws_vec(vec: Vec<u8>) -> anyhow::Result<Self> {
+        if vec.is_empty() {
+            bail!("error parsing relay::codec::Frame: too few bytes (0)");
+        }
+        let bytes = Bytes::from(vec);
+        let typ = FrameType::from(bytes[0]);
+        let frame = Self::from_bytes(typ, bytes.slice(1..))?;
+        Ok(frame)
+    }
+
+    fn into_ws_vec(self) -> Vec<u8> {
+        let mut bytes = BytesMut::new();
+        bytes.put_u8(self.typ().into());
+        self.write_to(&mut bytes);
+        bytes.to_vec()
```

Comment:
you can make this a bit nicer, by changing `fn write_to` to take `impl BufMut` instead of `BytesMut` as writer, that way you can pass in a `Vec`